### ISSUE TITLE
Why the lack of activation toggles under leave/instant "cleanable data"?

### ISSUE THREAD

**Issue by jm-ml** [Lusito/forget-me-not#176]
On the "Cleanable Data" page under cleanup in the settings, the "startup" option has many more checkboxes than "on leave", and both of them more than "instantly".  I've been trying to figure out why this is — I just don't get it. It seems like "on leave", for example, is erasing everything as I want it to, but why are  "plugin data", "indexed db", ect. not options for any but the "clean on startup" selection?   A brief explanation would be appreciated, I can't find any info elsewhere already addressed.

**Comment by Lusito** [comment]
The reason is simple: It's not supported by Firefox at this time.
In more detail:
Some of these cleanups can be done based on a domain.. e.g. "remove cookies from github.com".
Some can't. e.g. "remove indexeddb from github.com". There is only "remove indexeddb from all websites".

If there is no possibility to remove something by domain name, then there is no good way to clean it on domain leave. I could - in theory - delete all indexeddb when leaving a domain, but when some page is still open that uses indexeddb, it might break. That's why there is no checkbox for it.

Once firefox supports these types of cleanup, I will add them.

Here are some tickets you can vote for to improve the situation:
https://bugzilla.mozilla.org/show_bug.cgi?id=1340511
https://bugzilla.mozilla.org/show_bug.cgi?id=1551301
https://bugzilla.mozilla.org/show_bug.cgi?id=1380445
https://bugzilla.mozilla.org/show_bug.cgi?id=1526246
https://bugzilla.mozilla.org/show_bug.cgi?id=1353726
https://bugzilla.mozilla.org/show_bug.cgi?id=1464895

I guess I could write this information down somewhere, so I'll leave this ticket open as a reminder to create an FAQ entry or something.

**Comment by jm-ml** [comment]
Ok, thanks. I'm assuming then that with "on leave" selected as the default, the app takes no action regarding those other elements? If this is the case, a good feature would be the option to set different default actions for each category (i.e. — without any matching rule, cookies/storage/history will be deleted on leave, but plug-in data/cache/etc. may be independently set to be removed on startup.  Or is this effectively what is already happening?

**Comment by Lusito** [comment]
Not sure I get your first question correctly.. Maybe [this detailed explanation](https://github.com/Lusito/forget-me-not/issues/72#issuecomment-525855623) helps understanding the logic?

You can currently enable cleanups per category. For example:
- You set "On Leave" as fallback rule
- For cookies, you enabled the checkboxes "On Startup", but not "On Leave".
- Now cookies will only be cleaned on startup.

If I understood your use-case correctly, the above should do what you want.. if not, please give me a step-by-step use-case to follow.. like "I leave page x, I want y to happen, I restart my browser, I want z to happen, ...".

**Comment by jm-ml** [comment]
Yeah, that's what I wanted to know. I wasn't sure if the rule/cleanup settings worked as you described, or if the "On Startup" toggle settings, for example, only became relevant if  "Clean on Startup" was set as either a domain rule or the fallback rule.   I did notice on the extension info page that _conflicting_ rules would default to the most severe rule, but wasn't sure if a global "On Leave" rule would fallback to "On Startup" when On Leave was not an option.

**Comment by jingofett** [comment]
I believe this extension can clean indexedDB when leaving a site/domain:
https://addons.mozilla.org/en-US/firefox/addon/site-bleacher/

**Comment by Lusito** [comment]
No, that is still not possible on firefox. Sitebleacher can only do that on chrome. What will happen on firefox is all domains being cleaned by sitebleacher when one domain is left.
